src/modules/users/components/GasStation/GasStationPopover.tsx

```diff
@@ -38,8 +39,9 @@ const GasStationPopover = ({
     if (prevTxCount != null && txCount > prevTxCount) {
       setOpen(true);
       setTxNeedsSigning(true);
+      setUseCloseDelay(true);
```

Comment:
Not good enough. If the user creates a new transactions and then delays to sign it / approve it, the gas station will close.

See example:
![bug-gas-station-closes-prematurely](https://user-images.githubusercontent.com/1193222/179733460-db86505d-50ff-4caf-a767-9e9a054e63e1.gif)

You need a new check that takes into account the `txCount` and the `prevTxCount` while also checking if the transactions have succeeded, and only then close the gas station

Reply:
It will also auto-close if the transaction takes too long to mine.